pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -136,36 +168,66 @@ func buildOutboundNetworkFilters(env *model.Environment, node *model.Proxy, clus
 		}
+		tcpFilter = setAccessLogAndBuildTCPFilter(env, config)
+	}
+	return buildOutboundNetworkFiltersStack(port, tcpFilter, clusterName)
+}
 
-		if env.Mesh.AccessLogFile != "" {
-			fl := &fileaccesslog.FileAccessLog{
-				Path:   env.Mesh.AccessLogFile,
-				Format: EnvoyTCPLogFormat,
-			}
-			config.AccessLog = []*accesslog.AccessLog{
-				{
-					Config: util.MessageToStruct(fl),
-					Name:   xdsutil.FileAccessLog,
-				},
-			}
-		}
+// buildOutboundNetworkFiltersWithWeightedClusters takes a set of weighted
+// destination routes and builds a stack of network filters.
+func buildOutboundNetworkFiltersWithWeightedClusters(env *model.Environment, routes []*networking.RouteDestination,
+	push *model.PushContext, port *model.Port) []listener.Filter {
 
-		tcpFilter = &listener.Filter{
-			Name:   xdsutil.TCPProxy,
-			Config: util.MessageToStruct(config),
-		}
+	// construct TCP proxy using v2 config
+	statPrefix := buildStatPrefix(routes, port.Port, push)
+	clusterSpecifier := &tcp_proxy.TcpProxy_WeightedClusters{
+		WeightedClusters: &tcp_proxy.TcpProxy_WeightedCluster{},
+	}
+	config := &tcp_proxy.TcpProxy{
+		StatPrefix:       statPrefix,
+		ClusterSpecifier: clusterSpecifier,
+		// TODO: Need to set other fields such as Idle timeouts
+	}
+
+	for _, route := range routes {
+		clusterName := istio_route.GetDestinationCluster(route.Destination, push.ServiceByHostname[model.Hostname(route.Destination.Host)], port.Port)
+		clusterSpecifier.WeightedClusters.Clusters = append(clusterSpecifier.WeightedClusters.Clusters, &tcp_proxy.TcpProxy_WeightedCluster_ClusterWeight{
+			Name:   clusterName,
+			Weight: uint32(route.Weight),
+		})
 	}
 
+	tcpFilter := setAccessLogAndBuildTCPFilter(env, config)
+	return buildOutboundNetworkFiltersStack(port, tcpFilter, statPrefix)
+}
+
+// buildOutboundNetworkFiltersStack builds a slice of network filters based on
+// the protocol in use and the given TCP filter instance.
+func buildOutboundNetworkFiltersStack(port *model.Port, tcpFilter *listener.Filter, statPrefix string) []listener.Filter {
 	filterstack := make([]listener.Filter, 0)
 	switch port.Protocol {
 	case model.ProtocolMongo:
-		filterstack = append(filterstack, buildOutboundMongoFilter(clusterName))
+		filterstack = append(filterstack, buildOutboundMongoFilter(statPrefix))
 	}
 	filterstack = append(filterstack, *tcpFilter)
-
 	return filterstack
 }
 
+// buildOutboundNetworkFilters generates a TCP proxy network filter for outbound
+// connections. In addition, it generates protocol specific filters (e.g., Mongo
+// filter). This function constructs deprecated_v1 routes until the filter chain
+// match is ready.
+func buildOutboundNetworkFilters(env *model.Environment, node *model.Proxy,
+	routes []*networking.RouteDestination, push *model.PushContext,
+	deprecatedTCPFilterMatchAddress string, port *model.Port) []listener.Filter {
+
+	if (len(deprecatedTCPFilterMatchAddress) > 0 && !util.Is1xProxy(node)) || len(routes) == 1 {
```

Comment:
I think we can get rid of code with `util.Is1xProxy(...)`. Please confirm @rshriram.